From b095b4c38f01c73e2e055015e7c41367bc778790 Mon Sep 17 00:00:00 2001
From: Sergii Piatakov <sergii.piatakov@globallogic.com>
Date: Thu, 15 Nov 2018 15:21:34 +0200
Subject: [PATCH 1/2] improve calculation accuracy

Allow using float point arguments to avoid truncation.

Test: Add (4.9, 4.9) should return 10.
Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>

fix truncation error

To convert float to integer the truncation is performed, but the
rounding is expected.

Test: Add (4.9, 4.9) should return 10.
Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>
---
 calculator     |  1 +
 calculator.cpp | 22 +++++++++++-----------
 calculator.h   | 22 +++++++++++-----------
 3 files changed, 23 insertions(+), 22 deletions(-)
 create mode 160000 calculator

diff --git a/calculator b/calculator
new file mode 160000
index 0000000..4ad40a3
--- /dev/null
+++ b/calculator
@@ -0,0 +1 @@
+Subproject commit 4ad40a3b4572deee4efc716e7bd6e6e4fda46dcf
diff --git a/calculator.cpp b/calculator.cpp
index 3c63184..5432f6c 100644
--- a/calculator.cpp
+++ b/calculator.cpp
@@ -1,11 +1,11 @@
-#include "calculator.h"
-
-int Calculator::Add (int a, int b)
-{
-    return a + b;
-}
-
-int Calculator::Sub (int a, int b)
-{
-    return Add (a, -b);
-}
+#include "calculator.h"
+
+int Calculator::Add (double a, double b)
+{
+	return a + b + 0.5;
+}
+
+int Calculator::Sub (double a, double b)
+{
+    return Add (a, -b);
+}
diff --git a/calculator.h b/calculator.h
index 84ff414..44e1223 100644
--- a/calculator.h
+++ b/calculator.h
@@ -1,11 +1,11 @@
-#ifndef CALCULATOR_H
-#define CALCULATOR_H
-
-class Calculator
-{
-    public:
-        int Add (int, int);
-        int Sub (int, int);
-};
-
-#endif//CALCULATOR_H
+#ifndef CALCULATOR_H
+#define CALCULATOR_H
+
+class Calculator
+{
+    public:
+        int Add (double, double);
+        int Sub (double, double);
+};
+
+#endif//CALCULATOR_H
-- 
2.53.0

